code/obj/machinery/computer/operating.dm

```diff
@@ -26,56 +28,262 @@
+	if (damage >= O.max_damage)
+		return "Dead"
+	if (damage >= O.max_damage*0.9)
+		return "Critical"
+	if (damage >= O.max_damage*0.65)
+		return "Significant"
+	if (damage >= O.max_damage*0.3)
+		return "Moderate"
+	if (damage > 0)
+		return "Minor"
+	return "Okay"
+
+/obj/machinery/computer/operating/proc/generate_limb_data(var/mob/living/carbon/human/H)
+	var/list/limb_data = list()
+	var/current_status = ""
+	var/current_limb = ""
+	if (H.limbs)
+		current_limb = "Left Arm"
+		current_status = "Okay"
+		if (!H.limbs.l_arm)
+			current_status = "Missing"
+		else
+			if (istype(H.limbs.l_arm, /obj/item/parts/human_parts/arm/left/item))
+				var/obj/item/parts/human_parts/arm/left/item/I = H.limbs.l_arm
+				current_status = I.remove_object
+			else if (istype(H.limbs.l_arm, /obj/item/parts/robot_parts/arm/left/))
+				current_status = "Cybernetic"
+			else if (istype(H.limbs.l_arm, /obj/item/parts/artifact_parts/arm/))
+				current_status = "UNKNOWN"
+		limb_data += list(list(
+			"limb" = current_limb,
+			"status" = current_status,
+		))
+
+		current_limb = "Right Arm"
+		current_status = "Okay"
+		if (!H.limbs.r_arm)
+			current_status = "Missing"
+		else
+			if (istype(H.limbs.r_arm, /obj/item/parts/human_parts/arm/right/item))
+				var/obj/item/parts/human_parts/arm/right/item/I = H.limbs.r_arm
+				current_status = I.remove_object
+			else if (istype(H.limbs.r_arm, /obj/item/parts/robot_parts/arm/right))
+				current_status = "Cybernetic"
+			else if (istype(H.limbs.r_arm, /obj/item/parts/artifact_parts/arm/))
+				current_status = "UNKNOWN"
+		limb_data += list(list(
+			"limb" = current_limb,
+			"status" = current_status,
+		))
+
+		current_limb = "Left Leg"
+		current_status = "Okay"
+		if (!H.limbs.l_leg)
+			current_status = "Missing"
+		else
+			if (istype(H.limbs.l_leg, /obj/item/parts/robot_parts/leg/left))
+				current_status = "Cybernetic"
+			else if (istype(H.limbs.l_leg, /obj/item/parts/artifact_parts/leg/))
+				current_status = "UNKNOWN"
+		limb_data += list(list(
+			"limb" = current_limb,
+			"status" = current_status,
+		))
+
+		current_limb = "Right Leg"
+		current_status = "Okay"
+		if (!H.limbs.r_leg)
+			current_status = "Missing"
+		else
+			if (istype(H.limbs.r_leg, /obj/item/parts/robot_parts/leg/right))
+				current_status = "Cybernetic"
+			else if (istype(H.limbs.r_leg, /obj/item/parts/artifact_parts/leg/))
+				current_status = "UNKNOWN"
+		limb_data += list(list(
+			"limb" = current_limb,
+			"status" = current_status,
+		))
+
+		current_limb = "Butt" // look. okay. where else do i put it?
+		current_status = "Okay"
+		if(!H.organHolder?.butt)
+			current_status = "Missing"
+		else
+			if (istype(H.organHolder.butt, /obj/item/clothing/head/butt/cyberbutt))
+				current_status = "Cybernetic"
+		limb_data += list(list(
+			"limb" = current_limb,
+			"status" = current_status,
+		))
+
+	return limb_data
+
+/obj/machinery/computer/operating/proc/check_embedded_objects(var/mob/living/L)
```

Comment:
doesn't use src, it would be nice to use this for fill out scanprocs so they have a unified source?